app.py

```diff
@@ -1,42 +1,57 @@
+import os
 import streamlit as st
+from openai import OpenAI
+
+from dotenv import load_dotenv
+load_dotenv()
+client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
 
 def init_page():
-    '''stleamlit 初期化'''
+    """ ページ設定 """
     st.set_page_config(
-        page_title = 'Echo Bot'
+        page_title = "Echo Bot"
     )
-    st.title('Echo Bot')
-
-def init_chat():
-    if 'messages' not in st.session_state:
-        st.session_state['messages'] = []
-
-def print_chat_log():
-    '''chatの履歴を表示'''
-    for message in st.session_state.messages:
-        with st.chat_message(message['role']):
-            st.markdown(message['content'])
-
-def create_chat(prompt):
-    '''promptを読み取りbotが返信する'''
-    # 入力を出力
-    with st.chat_message('user'):
-        st.markdown(prompt)
-    # 入力したchatを履歴に追加
-    st.session_state.messages.append({'role':'user', 'content':prompt})
-    # botが返信
-    response = f'echo:{prompt}'
-    with st.chat_message('assistant'):
-        st.markdown(response)
-    # botの返答を履歴に追加
-    st.session_state.messages.append({'role':'assistant', 'content':response})
+    st.title("Echo Bot")
+    if "chat_log" not in st.session_state:
+        st.session_state.chat_log = [{"role": "system", "content": "あなたは日本のSES企業の事情に精通するスペシャリストです"}]
 
-def main():
-    init_chat()
-    print_chat_log()
-    if prompt := st.chat_input('What is up?'):
-        create_chat(prompt)
 
-if __name__ == '__main__':
+def get_llm_response(query):
+    """ LLMにクエリを送信し、回答を取得 """
+    messages = {"role": "user", "content": query}
+    # 履歴に追加(user)
+    st.session_state.chat_log.append(messages)
+    response = client.chat.completions.create(
+        model = "gpt-4o-mini",
+        messages = st.session_state.chat_log,
+        temperature = 0
+    )
+    answer = response.choices[0].message.content
+    return answer
+
+
+def chat_interface():
+    """ chat機能全般 """
+    # ログ表示
+    for message in st.session_state.chat_log[1:]:
+        with st.chat_message(message["role"]):
+            st.write(message["content"])
+
+    query = st.chat_input("質問を入力")
+    if query:
+        answer = get_llm_response(query)
+
+        with st.chat_message("user"):
+            st.write(query)
+
+        with st.chat_message("assistant"):
+            st.write(answer)
+        # 履歴に追加(assistant)
+        st.session_state.chat_log.append({"role": "assistant", "content": answer})
```

Comment:
- 履歴に追加はここでまとめますね（get_llm_res関数内でした方が良かったかも、後で修正）
```suggestion
        # 履歴に追加
        st.session_state.chat_log.append({"role": "user", "content": query})
        st.session_state.chat_log.append({"role": "assistant", "content": answer})
```